File: Cucumber/lib/pages/homepage.rb
class HomePage < SitePrism::Page
  set_url '/'

  elements :links, '#main-menu a'
  elements :section_links, '.section-links a'

  def goto destination
    links.each do |link|
      if destination == link.text
        link.click
        break
      end
    end
  end

  def links_count
    section_links.count
  end
end

File: Specflow/Specflow/Pages/HomePage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium;

namespace Specflow.Pages
{
    public class HomePage: BasePageModel
    {
        public HomePage(IWebDriver driver)
            : base(driver)
        {
        }

        #region Selector
        private readonly By _links = By.CssSelector("#main-menu a");
        private readonly By _records = By.CssSelector(".section-links a");


        #endregion

        #region Actions

        public void GoTo(string expectedLink)
        {
            var links = GetWebElements(_links);
            foreach (var link in links)
            {
                if (expectedLink.Equals(link.Text))
                {
                    link.Click();
                    break;
                }
            }
        }
        
        #endregion

        public int links_count()
        {
            return GetWebElements(_records).Count;
        }
    }
}
